src/main/java/com/internship/backend/service/ReservationService.java

```diff
@@ -0,0 +1,127 @@
+package com.internship.backend.service;
+
+import com.internship.backend.dto.ReservationDTO;
+import com.internship.backend.exceptions.ReservationAlreadyExists;
+import com.internship.backend.exceptions.ReservationDoesNotExistException;
+import com.internship.backend.model.NewDate;
+import com.internship.backend.model.Reservation;
+import com.internship.backend.model.TennisCourt;
+import com.internship.backend.model.Users;
+import com.internship.backend.repository.PriceRepository;
+import com.internship.backend.repository.ReservationRepository;
+import com.internship.backend.repository.TennisCourtRepository;
+import com.internship.backend.repository.UserRepository;
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ReservationService {
+
+    @Autowired
+    private ReservationRepository reservationRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private TennisCourtRepository tennisCourtRepository;
+
+    @Autowired
+    private PriceRepository priceRepository;
+
+    public List<Reservation> getAllReservations(){
+        return reservationRepository.findAll();
+    }
+
+    public Reservation addReservation(Reservation reservation) throws ReservationAlreadyExists {
+
+        if (isValidReservation((reservation)))
+        {
+            return reservationRepository.save(reservation);
+        }
+        else{
+            throw new ReservationAlreadyExists("Reservation intersects another one");
+        }
+    }
+
+    public boolean isValidReservation(Reservation reservation){
+        List<Reservation> reservationList = reservationRepository.findAll();
+        for(Reservation elem : reservationList){
+            if((reservation.getStartTime().isAfter(elem.getEndTime()) || reservation.getStartTime().equals(elem.getEndTime()))
+            && (reservation.getEndTime().isBefore(elem.getStartTime()) || reservation.getEndTime().equals(elem.getStartTime()))){
+                return true;
+            }
+            else return false;
+        }
+        return true;
```

Comment:
If possible avoid methods returning multiple values. Could this method be simplified in order to have like two return statements?